### ISSUE TITLE
Gradient of the norm of the zero tangent vector on the SPD manifold is NaN

### ISSUE THREAD

**Issue by leo-buecher** [geoopt/geoopt#243]
Hi, the backward pass of the norm function of the SPD manifold (with Affine Invariant metric) returns NaN when the input tangent vector is zero. It is mathematically understandable, but it could be great that it behaves like torch.norm and returns the zero subgradient.

**To Reproduce**
```
import torch
import geoopt

X = torch.eye(2)
U = torch.zeros((2, 2))
U.requires_grad_(True)

spd_man = geoopt.SymmetricPositiveDefinite()
norm = spd_man.norm(X, U)
norm.backward()
print(U.grad)
```
prints
```
tensor([[nan, nan],
        [nan, nan]])
```

**Expected behavior**
It would be great that it returns the zero vector, similarly to torch.norm.

**Complementary information**
I use pytorch version 2.5.1 and geoopt version 0.5.1, installed with pip.

Thank you !
